src/widgets/helper/ChannelView.cpp

```diff
@@ -2892,17 +2754,17 @@ void ChannelView::selectWholeMessage(MessageLayout *layout, int &messageIndex)
     this->setSelection(msgStart, msgEnd);
 }
 
-void ChannelView::getWordBounds(MessageLayout *layout,
-                                const MessageLayoutElement *element,
-                                const QPoint &relativePos, int &wordStart,
-                                int &wordEnd)
+std::pair<size_t, size_t> ChannelView::getWordBounds(
+    MessageLayout *layout, const MessageLayoutElement *element,
+    const QPoint &relativePos)
 {
-    const int mouseInWordIndex = element->getMouseOverIndex(relativePos);
-    wordStart = layout->getSelectionIndex(relativePos) - mouseInWordIndex;
+    const auto wordStart = layout->getSelectionIndex(relativePos) -
+                           element->getMouseOverIndex(relativePos);
     const int selectionLength = element->getSelectionIndexCount();
     const int length =
         element->hasTrailingSpace() ? selectionLength - 1 : selectionLength;
-    wordEnd = wordStart + length;
+
+    return {wordStart, wordStart + length};
```

Comment:
I wonder if we can use `size_t` for everything here without running into overflow problems.

Reply:
everything will use size_t now ce10b811b024fe625c0f7b8c3f997b7c3aff659f